src/modules/cmdpal/Microsoft.CmdPal.UI/ShellPage.xaml.cs

```diff
@@ -148,7 +185,27 @@ public void Receive(ShowDetailsMessage message)
 
     public void Receive(HideDetailsMessage message) => HideDetails();
 
+    public void Receive(LaunchUriMessage message) => _ = Launcher.LaunchUriAsync(message.Uri);
+
+    public void Receive(HandleCommandResultMessage message) => HandleCommandResult(message.Result.Unsafe);
+
     private void HideDetails() => ViewModel.IsDetailsVisible = false;
 
-    public void Receive(LaunchUriMessage message) => _ = Launcher.LaunchUriAsync(message.Uri);
+    private void GoBack()
+    {
+        HideDetails();
+        RootFrame.GoBack();
+        RootFrame.ForwardStack.Clear();
+        SearchBox.Focus(Microsoft.UI.Xaml.FocusState.Programmatic);
+    }
+
+    private void GoHome()
+    {
+        while (RootFrame.CanGoBack)
+        {
+            GoBack();
+        }
+
+        SearchBox.ClearSearch();
```

Comment:
Since other things need to respond to "going home" we should probably make this a message that gets sent out.

Then this class itself can listen the message in making this a receive event (and we send the message in the instead of calling GoHome above elsewhere).

And then the SearchBox can also listen to the GoHome message and then just internally clear it's search filter vs. exposing a new public method.

Reply:
Easy enough to me. I can whip that up